worlds/timespinner/Options.py

```diff
@@ -388,67 +357,256 @@ class TrapChance(Range):
-    "HpCap": HpCap,
-    "LevelCap": LevelCap,
-    "ExtraEarringsXP": ExtraEarringsXP,
-    "BossHealing": BossHealing,
-    "ShopFill": ShopFill,
-    "ShopWarpShards": ShopWarpShards,
-    "ShopMultiplier": ShopMultiplier,
-    "LootPool": LootPool,
-    "DropRateCategory": DropRateCategory,
-    "FixedDropRate": FixedDropRate,
-    "LootTierDistro": LootTierDistro,
-    "ShowBestiary": ShowBestiary,
-    "ShowDrops": ShowDrops,
-    "EnterSandman": EnterSandman,
-    "DadPercent": DadPercent,
-    "RisingTides": RisingTides,
-    "RisingTidesOverrides": RisingTidesOverrides,
-    "UnchainedKeys": UnchainedKeys,
-    "TrapChance": TrapChance,
-    "Traps": Traps,
-    "PresentAccessWithWheelAndSpindle": PresentAccessWithWheelAndSpindle,
-    "DeathLink": DeathLink,
-}
-
-
-def is_option_enabled(world: MultiWorld, player: int, name: str) -> bool:
-    return get_option_value(world, player, name) > 0
-
-
-def get_option_value(world: MultiWorld, player: int, name: str) -> Union[int, Dict, List]:
-    option = getattr(world, name, None)
-    if option == None:
-        return 0
-
-    return option[player].value
+    display_name = "Back to the future"
+
+@dataclass
+class TimespinnerOptions(PerGameCommonOptions, DeathLinkMixin):
+    start_with_jewelry_box: StartWithJewelryBox
+    downloadable_items: DownloadableItems
+    eye_spy: EyeSpy
+    start_with_meyef: StartWithMeyef
+    quick_seed: QuickSeed
+    specific_keycards: SpecificKeycards
+    inverted: Inverted
+    gyre_archives: GyreArchives
+    cantoran: Cantoran
+    lore_checks: LoreChecks
+    boss_rando: BossRando
+    damage_rando: DamageRando
+    damage_rando_overrides: DamageRandoOverrides
+    hp_cap: HpCap
+    level_cap: LevelCap
+    extra_earrings_xp: ExtraEarringsXP
+    boss_healing: BossHealing
+    shop_fill: ShopFill
+    shop_warp_shards: ShopWarpShards
+    shop_multiplier: ShopMultiplier
+    loot_pool: LootPool
+    drop_rate_category: DropRateCategory
+    fixed_drop_rate: FixedDropRate
+    loot_tier_distro: LootTierDistro
+    show_bestiary: ShowBestiary
+    show_drops: ShowDrops
+    enter_sandman: EnterSandman
+    dad_percent: DadPercent
+    rising_tides: RisingTides
+    rising_tides_overrides: RisingTidesOverrides
+    unchained_keys: UnchainedKeys
+    back_to_the_future: PresentAccessWithWheelAndSpindle
+    trap_chance: TrapChance
+    traps: Traps
+
+class HiddenDamageRandoOverrides(DamageRandoOverrides): 
+    """Manual +/-/normal odds for an orb. Put 0 if you don't want a certain nerf or buff to be a possibility. Orbs that
+    you don't specify will roll with 1/1/1 as odds"""
+    visibility = Visibility.none
+
+class HiddenRisingTidesOverrides(RisingTidesOverrides):
+    """Odds for specific areas to be flooded or drained, only has effect when RisingTides is on.
+    Areas that are not specified will roll with the default 33% chance of getting flooded or drained"""
+    visibility = Visibility.none
+
+class HiddenTraps(Traps):
```

Comment:
I'm wondering why these are fully hidden, these certainly sound like options users would want to actually use